src/interpreter/initialization.rs

```diff
@@ -109,35 +104,32 @@ impl<S> Interpreter<S> {
         // Reduce by unavailable balances
         let base_asset = AssetId::default();
         if let Some(base_asset_balance) = balances.get_mut(&base_asset) {
-            // remove byte costs from base asset spendable balance
-            let byte_balance = (tx.metered_bytes_size() as Word)
-                .checked_mul(tx.byte_price())
-                .ok_or(ValidationError::ArithmeticOverflow)?;
-
-            // remove gas costs from base asset spendable balance
-            // gas = limit * price
-            let gas_cost = tx
-                .gas_limit()
-                .checked_mul(tx.gas_price())
-                .ok_or(ValidationError::ArithmeticOverflow)?;
-
-            // add up total amount of required fees
-            let total_fee = byte_balance
-                .checked_add(gas_cost)
-                .ok_or(ValidationError::ArithmeticOverflow)?;
+            // calculate the fee with used witness bytes + gas limit
```

Comment:
[nit] Witnesses aren't included in the metered byte size, otherwise validators could add an arbitrary number of witnesses to each tx to max out the fee.

```suggestion
            // calculate the fee with used metered bytes + gas limit
```